index.d.ts

```diff
@@ -1,19 +1,79 @@
 declare namespace pug {
-  export function compile (str: string, options?: Options): Template;
-  export function compileFile (filename: string, options?: Options): Template;
-  export function render (str: string, options?: Options & any, fn?: (err: Error | void, res: string) => void): string;
-  export function renderFile (filename: string, options?: Options & any, fn?: (err: Error | void, res: string) => void): string;
-
-  export interface Options {
-    cache?: boolean;
-    filename?: string;
-    compileDebug?: boolean;
-    pretty?: boolean;
-  }
-
-  export interface Template {
-    (locals?: any): string;
-  }
+    /**
+     * Compile a Pug template to a function which can be rendered multiple times with different locals.
+     */
+    export function compile(template: string, options?: Options): TemplateFunction;
+
+    /**
+     * Compile a Pug template from a file to a function which can be rendered multiple times with different locals.
+     */
+    export function compileFile(path: string, options?: Options): TemplateFunction;
+
+    /**
+     * Compile a Pug template to a string of JavaScript that can be used client side along with the Pug runtime.
+     */
+    export function compileClient(template: string, options?: Options): string;
+
+    /**
+     * Compile a Pug template to an object of the form:
+     * {
+     *   'body': 'function (locals) {...}',
+     *   'dependencies': ['filename.pug']
+     * }
+     * that can be used client side along with the Pug runtime.
+     * You should only use this method if you need dependencies to implement something like watching for changes to the Pug files.
+     */
+    export function compileClientWithDependenciesTracked(template: string, options?: any): {
```

Comment:
It's the same, I forgot to change it.
Checked in https://github.com/pugjs/pug/blob/master/lib/index.js.